src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -167,6 +176,41 @@ public EngineConfig(ShardId shardId, boolean optimizeAutoGenerateId, ThreadPool
         failEngineOnCorruption = indexSettings.getAsBoolean(INDEX_FAIL_ON_CORRUPTION_SETTING, true);
         failOnMergeFailure = indexSettings.getAsBoolean(INDEX_FAIL_ON_MERGE_FAILURE_SETTING, true);
         gcDeletesInMillis = indexSettings.getAsTime(INDEX_GC_DELETES_SETTING, EngineConfig.DEFAULT_GC_DELETES).millis();
+        versionMapSizeSetting = indexSettings.get(INDEX_VERSION_MAP_SIZE, DEFAULT_VERSION_MAP_SIZE);
+        updateVersionMapSize();
+    }
+
+    /** updates {@link #versionMapSize} based on current setting and {@link #indexingBufferSize} */
+    private void updateVersionMapSize() {
```

Comment:
I don't think it will matter? I tried to keep this as simple as possible. This shouldn't be used as a way to reduced size quickly. On top of that - a refresh will trim it down anyway, no?